packages/react-inbox/src/reducer.ts

```diff
@@ -64,8 +64,27 @@ export default (state: InboxState = {}, action) => {
       startCursor: action?.payload?.startCursor,
       messages: action?.payload?.appendMessages ? [
         ...(state.messages ??[]),
-        ...newMessages
-      ] : newMessages
+        ...newMessages,
+      ] : newMessages,
+    };
+  }
+
+  case "inbox/MARK_MESSAGE_UNREAD": {
+    const unreadMessageCount = (state.unreadMessageCount ?? 1) + 1;
+
+    return {
+      ...state,
+      messages: state.messages?.map(message => {
```

Comment:
why are we not sure that messages are legit array? When we dispatch an action, we would have sensible default / initial state right?

Reply:
Fixed.